packages/core/src/dependency/TransitiveDependencyResolver.ts

```diff
@@ -0,0 +1,114 @@
+
+import ProjectConfig from '../project/ProjectConfig';
+import { COLOR_HEADER, COLOR_KEY_MESSAGE, COLOR_SUCCESS, COLOR_ERROR } from '@dxatscale/sfp-logger';
+import QueryHelper from '../queryHelper/QueryHelper';
+import { Connection } from '@salesforce/core';
+const Table = require('cli-table');
+
+
+export default class TransitiveDependencyResolver {
+
+  private dependencyMap;
+  private updatedprojectConfig: any
+
+  constructor(
+    private projectConfig: ProjectConfig,
+    private conn: Connection 
+    ){}
+  public async exec(): Promise<ProjectConfig>{
+    console.log('Validating Project Dependencies...')
+
+    this.updatedprojectConfig = this.projectConfig
+
+    this.dependencyMap = await this.getAllPackageDependencyMap(this.projectConfig);
+
+    await this.validateDependency(this.dependencyMap, this.projectConfig)
+
+    return this.updatedprojectConfig
+     
+  }
+
+  private getAllPackageDependencyMap(projectConfig: any): any {
+    let pkgWithDependencies = {}
+    let packages = projectConfig.packageDirectories
+    for(let pkg of packages){
+      if(pkg.dependencies){
+        pkgWithDependencies[pkg.package] = pkg.dependencies
+      }
+    }
+    return pkgWithDependencies
+  }
+
+  private async validateDependency(dependencyMap: any, projectConfig: any){
+    let pkgs = Object.keys(dependencyMap)
+    for ( let pkg of pkgs){
+      console.log(
```

Comment:
Replace console logs with SFPLogger and adequate loglevels  also look into the targer such as file or console